### PR TITLE
Update UCAS matches dual application resolved on UCAS emails

### PR DESCRIPTION
**Pull request by AgaDufrat**
## Context

We need better copy.

## Changes proposed in this pull request

- Don't use translations to test UCAS matches emails subjects
- Add missing email previews for resolved on UCAS and Apply
- Send resolved on UCAS automatically email when we requested withdrawal from UCAS
- Update email copy for UCAS match resolved on UCAS

## Guidance to review

See mailer previews

## Link to Trello card

https://trello.com/c/j9YmgCWM/3276-update-ucas-matches-emails-content-dual-applications

## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/services/ucas_matches/resolve_on_ucas.rb [hunk 1]
```diff
@@ -7,8 +7,10 @@ def initialize(ucas_match)
     end
 
     def call
+      automatic_withdrawal ||= ucas_match.ucas_withdrawal_requested?
```
Comment (duncanjbrown) [inline app/services/ucas_matches/resolve_on_ucas.rb:10]: Could `automatic_withdrawal` ever have been initialised?

**Review by duncanjbrown — COMMENTED** [review]
Code looks good! I wonder if `automatically` is the right word here. We send this email if Apply staff have written to UCAS requesting withdrawal and the application has subsequently been resolved on UCAS. Is it more like `ucas_match_resolved_on_ucas_at_our_request`?

Taken together, do the tests for `ResolveOnUCAS` and `SendResolvedOnUCASEmails` feel a bit stubby? Likely to have to change if we change details of the implementation. Would it be possible to perhaps have a test in `ResolveOnUCAS` which went all the way to checking the right emails were sent?

File: app/services/ucas_matches/resolve_on_ucas.rb (hunk 1 above)
Comment (AgaDufrat) [reply app/services/ucas_matches/resolve_on_ucas.rb:10]: I decided to do it this way to access the previous value of `action_taken` as it gets updated to `resolved_on_ucas` on line 12. 
On reflection, I should probably use `action_taken_was == 'ucas_withdrawal_requested'`

File: app/services/ucas_matches/send_resolved_on_ucas_emails.rb [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
+      @at_our_request = at_our_request
     end
 
     def call
       raise 'The application has not been resolved on UCAS' if !ucas_match.resolved_on_ucas?
 
       application_choice = ucas_match.application_choices_for_same_course_on_both_services.first
 
-      CandidateMailer.ucas_match_resolved_on_ucas_email(application_choice).deliver_later
+      if at_our_request
```
Comment (duncanjbrown) [inline app/services/ucas_matches/send_resolved_on_ucas_emails.rb:15]: Very nice!

File: spec/services/ucas_matches/send_resolved_on_ucas_emails_spec.rb [hunk 3]
```diff
@@ -27,4 +27,23 @@
       expect(ProviderMailer).to have_received(:ucas_match_resolved_on_ucas_email).with(provider_user, application_choice)
     end
   end
+
+  context 'when the application has been resolved on ucas automatically' do
```
Comment (duncanjbrown) [inline spec/services/ucas_matches/send_resolved_on_ucas_emails_spec.rb:31]:
```suggestion
  context 'when the application has been resolved on ucas at our request' do
```

**Review by duncanjbrown — APPROVED** [review]
Thanks for changes @AgaDufrat !

**Pull request merged**